src/module/macros/activation.ts

```diff
@@ -0,0 +1,168 @@
+import { renderMacroHTML } from "./render"
+import { rollTechMacro } from "./tech"
+
+const lp = LANCER.log_prefix;
+
+
+export async function prepareActivationMacro(
+  a: string,
+  i: string,
+  type: ActivationOptions,
+  index: number,
+  rerollData?: AccDiffDataSerialized
+) {
+  // Determine which Actor to speak as
+  let actor = getMacroSpeaker(a);
+  if (!actor) return;
+
+  // Get the item
+  let item: LancerItem | undefined;
+  item = actor.items.get(i);
+  if (!item && actor.is_mech()) {
+    let pilot = game.actors!.get(actor.data.data.pilot?.id ?? "");
+    item = pilot?.items.get(i);
+  }
+
+  if (!item || (!actor.is_mech() && !actor.is_pilot())) {
+    return ui.notifications!.error(
+      `Error preparing tech attack macro - could not find Item ${i} owned by Actor ${a}! Did you add the Item to the token, instead of the source Actor?`
+    );
+  } else if (!item.isOwned) {
+    return ui.notifications!.error(`Error rolling tech attack macro - ${item.name} is not owned by an Actor!`);
+  } else if (!item.is_mech_system() && !item.is_npc_feature() && !item.is_talent()) {
+    return ui.notifications!.error(`Error rolling tech attack macro - ${item.name} is not a System or Feature!`);
+  }
+
+  let itemEnt: MechSystem | NpcFeature | Talent = await item.data.data.derived.mm_promise;
+  let actorEnt: Mech | Pilot = await actor.data.data.derived.mm_promise;
+
+  if (getAutomationOptions().limited_loading && is_tagged(itemEnt) && is_limited(itemEnt) && itemEnt.Uses <= 0) {
+    ui.notifications!.error(`Error using item--you have no uses left!`);
+    return;
+  }
+
+  // TODO--handle NPC Activations
+  if (itemEnt.Type === EntryType.NPC_FEATURE) return;
+
+  switch (type) {
+    case ActivationOptions.ACTION:
+      switch (itemEnt.Actions[index].Activation) {
+        case ActivationType.FullTech:
+        case ActivationType.Invade:
+        case ActivationType.QuickTech:
+          let partialMacroData = {
+            title: "Reroll activation",
+            fn: "prepareActivationMacro",
+            args: [a, i, type, index],
+          };
+          _prepareTechActionMacro(actorEnt, itemEnt, index, partialMacroData, rerollData);
+          break;
+        default:
+          _prepareTextActionMacro(actorEnt, itemEnt, index);
+      }
+    case ActivationOptions.DEPLOYABLE:
+      _prepareDeployableMacro(actorEnt, itemEnt, index);
+  }
+
+  // Wait until the end to deduct a use so we're sure it completed succesfully
+  if (getAutomationOptions().limited_loading && is_tagged(itemEnt) && is_limited(itemEnt)) {
+    itemEnt.Uses = itemEnt.Uses - 1;
+    await itemEnt.writeback();
+  }
+
+  return;
+}
+
+async function _prepareTextActionMacro(
+  actorEnt: Mech | Pilot | Npc,
+  itemEnt: Talent | MechSystem | NpcFeature,
+  index: number
+) {
+  // Support this later...
+  // TODO: pilot gear and NPC features
+  if (itemEnt.Type !== EntryType.MECH_SYSTEM && itemEnt.Type !== EntryType.TALENT) return;
```

Comment:
Note for future - I think we have some `is_reg_x()` helper functions for uses like this. Unless we only have those for the actor types.